python/pyspark/sql/session.py

```diff
@@ -405,7 +401,7 @@ def _createFromLocal(self, data, schema):
             data = list(data)
 
         if schema is None or isinstance(schema, (list, tuple)):
-            struct = self._inferSchemaFromList(data)
+            struct = self._inferSchemaFromList(data, names=schema)
             converter = _create_converter(struct)
             data = map(converter, data)
             if isinstance(schema, (list, tuple)):
```

Comment:
@gberger, could we remove this branch too?

Reply:
yes

Reply:
removed with commit 5131db2